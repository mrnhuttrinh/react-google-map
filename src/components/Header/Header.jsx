import React from 'react';
import block from 'bem-cn';

import './Header.scss';

const b = block('header');

class Header extends React.PureComponent {
  render() {
    return (
      <header className={b()}>
        <div className={b('top')}>
          <div className={b('top--logo')}>
            <img src={require('./imgs/logo.PNG')} />
          </div>
          <div className={b('top--menu')}>
            <ul className={b('top--menu_left')}>
              <li className="active">Overview</li>
              <li>Fleet Utilisation</li>
              <li>Violations</li>
              <li>Drivers Scoreboard</li>
              <li>Vehicle Maintenance</li>
              <li>Reports</li>
            </ul>
            <ul className={b('top--menu_right')}>
              <li><i className={b('icon').mix('far fa-user')}></i></li>
              <li><i className={b('icon').mix('fas fa-sign-out-alt')}></i></li>
            </ul>
          </div>
        </div>
        <div className={b('bottom')}>
          <div>
            &#8592; Back to Overview
          </div>
          <div>
            Connect4carâ€™s Fleet Summary
          </div>
        </div>
      </header>
    );
  }
}


export default Header;