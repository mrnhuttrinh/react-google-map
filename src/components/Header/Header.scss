.header {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 180px;

  &__top {
    display: flex;
    width: 100%;
    height: 60px;

    &--logo {
      width: 120px;
      display: flex;
      justify-content: center;
      align-items: center;

      > img {
        width: 50px;
        height: 50px;
      }
    }

    &--menu {
      margin-left: 20px;
      margin-right: 5px;
      display: flex;
      flex: 1;
      justify-content: space-between;
      border-bottom: 1px solid #0071ca;

      &_left {
        display: flex;
        list-style: none;
        height: 100%;
        line-height: 100%;
        align-items: center;
        text-decoration: none;

        > li {
          display: inline-block;
          cursor: pointer;
          padding-right: 10px;
          padding-left: 10px;
          font-size: 14px;
          font-style: normal;
          font-stretch: normal;
          line-height: normal;
          letter-spacing: 0.2px;
          color: #8c8aa2;
          line-height: 57px;

          &:hover {
            text-shadow: 1px 0px 0px #0071ca;
            border-bottom: 3px solid #0071ca;
          }
        }

        .active {
          text-shadow: 1px 0px 0px #0071ca;
          border-bottom: 3px solid #0071ca;
        }
      }

      &_right {
        display: flex;
        list-style: none;
        height: 100%;
        line-height: 100%;
        align-items: center;
        text-decoration: none;

        > li {
          display: inline-block;
          cursor: pointer;
          margin-right: 10px;
          margin-left: 10px;
          font-size: 14px;
          font-style: normal;
          font-stretch: normal;
          line-height: normal;
          letter-spacing: 0.2px;
          color: #0071ca;
        }
      }
    }
  }

  &__bottom {
    width: 100%;
    height:120px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 0 80px;

    > div {
      margin: 5px 0;

      &:first-child {
        text-decoration: underline;
      }

      &:last-child {
        font-size: 20px;
        font-weight: bold;
      }
    }
  }

  &__icon {
    font-size: 20px;
  }
}
