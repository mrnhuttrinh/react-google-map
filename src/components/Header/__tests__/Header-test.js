import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { MemoryRouter as Router, withRouter } from 'react-router-dom';
import { equal } from 'assert';
import { shallow } from 'enzyme';

import Header, { HeaderComponent } from '../Header';

const mockStore = configureStore();

const history = {
  push: jest.fn(),
}

const HeaderWithRouter = withRouter(Header);

const categories = [
  {
    href: "/all-stores/dining",
    id: 0,
    name: "Dining",
    priority: 1,
    shortname: "dining",
  }
];

const initialState = {
  categories: {
    categories: categories,
    stores: [],
  },
  ui: {
    showAll: false,
  },
};

test('should render Header component', () => {

  const store = mockStore(initialState)
  const component = renderer.create(
    <Provider store={store} history={history}>
      <Router>
        <HeaderWithRouter />
      </Router>
    </Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
  const instance = component.root;
  equal(instance.children.length, 1);
});

test('should try click category item', () => {
  const actions = {
    updateCategories: jest.fn(),
  };
  const wrapper = shallow(
    <HeaderComponent
      actions={actions}
      categories={categories}
    />
  );
  wrapper.find('NavLink').simulate('click');
  expect(actions.updateCategories).toHaveBeenCalled();
});
