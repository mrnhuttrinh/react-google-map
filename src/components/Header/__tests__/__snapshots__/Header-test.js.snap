// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component 1`] = `
<header
  className="header"
>
  <a
    aria-current={null}
    className="header__item"
    href="/all-stores/dining"
    onClick={[Function]}
    style={
      Object {
        "order": undefined,
      }
    }
  >
    <i
      className="icon fas fa-fire"
      style={
        Object {
          "color": "#ffc125",
        }
      }
    />
    <span>
      Dining
    </span>
  </a>
</header>
`;
